addons/sale_edi_ubl/models/sale_edi_common.py

```diff
@@ -0,0 +1,126 @@
+            # To do: rename tax_id field to `tax_ids` of sale.order.line
+            line_values['tax_id'], tax_logs = self._retrieve_taxes(
+                order, line_values, 'sale',
+            )
+            lines_values += self._retrieve_line_charges(order, line_values, line_values['tax_id'])
+            if not line_values['product_uom']:
+                line_values.pop('product_uom')  # if no uom, pop it so it's inferred from the product_id
+            lines_values.append(line_values)
+        return lines_values, logs + tax_logs
+
+    def _import_payment_term_id(self, order, tree, xapth):
+        """ Return payment term from given tree. """
+        payment_term_note = self._find_value(xapth, tree)
+        if not payment_term_note:
+            return False
+
+        return self.env['account.payment.term'].search([
+            *self.env['account.journal']._check_company_domain(order.company_id),
+            ('name', '=', payment_term_note)
+        ], limit=1)
+
+    def _import_delivery_partner(self, order, name, phone, email):
+        """ Import delivery address from details if not found then log details."""
+        logs = []
+        dest_partner = self.env['res.partner'] \
+            .with_company(order.company_id) \
+            ._retrieve_partner(name=name, phone=phone, email=email)
+        if not dest_partner:
+            partner_detaits_str = self._get_partner_detail_str(name, phone, email)
+            logs.append(_("Could not retrive Delivery Address with Details: { %s }", partner_detaits_str))
+
+        return dest_partner, logs
+
+    def _import_partner(self, company_id, name, phone, email, vat,
+        country_code=False,
+        peppol_eas=False,
+        peppol_endpoint=False,
+    ):
+        """ Override of edi.mixin to set current user partner if there is no matching partner
+        found and log details related to partner."""
+        partner, logs = super()._import_partner(
+            company_id, name, phone, email, vat,
+            country_code=country_code,
+            peppol_eas=peppol_eas,
+            peppol_endpoint=peppol_endpoint,
+        )
+        if not partner:
+            partner_detaits_str = self._get_partner_detail_str(name, phone, email, vat)
+            logs.append(_("Could not retrive Customer with Details: { %s }", partner_detaits_str))
+        return partner, logs
+
+    def _get_partner_detail_str(self, name=False, phone=False, email=False, vat=False):
+        """ Return partner details string to help user find or create proper contact with details.
+        """
+        partner_details = ""
+        if name:
+            partner_details += "\n Name: %s" % name
+        if phone:
+            partner_details += ",\n Phone: %s" % phone
+        if email:
+            partner_details += ",\n Email: %s" % email
+        if vat:
+            partner_details += ",\n Vat: %s" % vat
```

Comment:
translation for non english speakers ?

Reply:
Already added translation for whole string where it used :see_no_evil: 
Do we need to add translation in each sub string to make it properly work I think I need to add kw string instead of %s to not have weird string :thinking:

Reply:
```python
a = 'hello world'
b = 'weather's nice today'

c = _("%s %s", a, b) # Doesn't work

a = _("hello world")
b = _("weather's nice today")
c = _("%s %s", a, b) # works, but doesn't allow translators to fully reorganize the sentence according to language specificities
c = _("%(a)s %(b)s", a=a, b=b)  # more control, translators can move b before a if needed, but cannot split the content of a or b.

c = _("hello world weather's nice today")  # Gives translators full control on the whole sentence
```

More details and explanation [here](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#use-translation-method-correctly)

Reply:
Got it! Thank you for detailed explanation :smile: